src/spec.c

```diff
@@ -2605,7 +2605,11 @@ void Indexes_ReplaceMatchingWithSchemaRules(RedisModuleCtx *ctx, RedisModuleStri
       // on the spec from section.
       continue;
     }
-    IndexSpec_UpdateDoc(specOp->spec, ctx, to_key, getDocTypeFromString(to_key));
+    DocumentType type = getDocTypeFromString(to_key);
+    if (type == DocumentType_Unsupported) {
+      continue;
+    }
```

Comment:
We can check this at the beginning of the function and return if the type is not supported.